psydac/core/bsplines.py

```diff
@@ -384,6 +392,8 @@ def histopolation_matrix(knots, degree, periodic, normalization, xgrid, check_bo
     if not np.all(np.diff(xgrid) > 0):
         raise ValueError("Grid points must be ordered, with no repetitions: {}".format(xgrid))
 
+    knots = np.asarray(knots)
+    xgrid = np.asarray(xgrid)
```

Comment:
As above

Reply:
Done in https://github.com/pyccel/psydac/commit/54fa1fb4c2afba79c77392308a193bad59161fee